lint.go

```diff
@@ -1159,11 +1159,17 @@ func (f *file) lintErrors() {
 	}
 }
 
-func lintCapAndPunct(s string) (isCap, isPunct bool) {
+func lintCapAndPunct(s string) (isCap, isPunct, isNewline bool) {
 	first, firstN := utf8.DecodeRuneInString(s)
 	last, _ := utf8.DecodeLastRuneInString(s)
 	isPunct = last == '.' || last == ':' || last == '!'
+	isNewline = last == '\n'
 	isCap = unicode.IsUpper(first)
+	if isNewline && strings.Index(s, "\n") != len(s)-1 {
+		// Don't flag strings that contain more than one newline. They
```

Comment:
I believe we should flag errors ending with a newline regardless of whether is a multiline message or not.

Here's my rationale:
* Many of the violations for ending with a newline already violates the isCap rule. So they are already bad style.
* Ending with a newline makes printing of arbitrary errors inconsistent. That is, if I print some error `err`, I may or may not end up with an extra blank line depending on how the underlying error is formatted.
* It seems to me that the cases where trailing newlines were used, should have instead included a newline at the point where the error message was printed.